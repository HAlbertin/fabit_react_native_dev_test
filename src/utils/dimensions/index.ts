import { Dimensions } from 'react-native';

const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const DimensionUtils = {
  deviceWidth,
  deviceHeight,
};

export default DimensionUtils;
