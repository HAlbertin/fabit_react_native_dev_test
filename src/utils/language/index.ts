import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import StorageUtils from '../storage';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

export const getCurrentLocale = () => Localization.locale;

export const DEFAULT_LANGUAGE = 'en';
export const DEFAULT_TRANSLATION = {
  key: DEFAULT_LANGUAGE,
  value: 'English',
};

export const getCurrentLanguage = () => StorageUtils.getItem('USER_LANGUAGE') ?? DEFAULT_LANGUAGE;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);

export const LANGUAGES = [
  { lang: 'pt', file: pt, name: 'Language_Pt' },
  { lang: 'en', file: en, name: 'Language_En' },
  { lang: 'es', file: es, name: 'Language_Es' },
  { lang: 'it', file: it, name: 'Language_It' },
  { lang: 'de', file: de, name: 'Language_De' },
  { lang: 'fr', file: fr, name: 'Language_Fr' },
];

const setupInitialLanguage = () => {
  i18n.fallbacks = true;

  // Set the key-value pairs for the different languages you want to support.
  i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
    translations[lang] = file;
    return translations;
  }, {});

  // Set the locale once at the beginning of your app.
  i18n.locale = getCurrentLocale();
};

setupInitialLanguage();
