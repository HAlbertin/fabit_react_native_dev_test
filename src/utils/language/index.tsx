import * as Localization from 'expo-localization';
import i18n from 'i18n-js';
import de from './translations/de';
import en from './translations/en';
import es from './translations/es';
import fr from './translations/fr';
import it from './translations/it';
import pt from './translations/pt';
import { TranslationKeys } from './translations/translations.interface';

const LANGUAGES = [
  { lang: 'pt', file: pt },
  { lang: 'en', file: en },
  { lang: 'es', file: es },
  { lang: 'it', file: it },
  { lang: 'de', file: de },
  { lang: 'fr', file: fr },
];

// Set the key-value pairs for the different languages you want to support.
i18n.translations = LANGUAGES.reduce((translations, { lang, file }) => {
  translations[lang] = file;
  return translations;
}, {});

// Set the locale once at the beginning of your app.
i18n.locale = Localization.locale;

export const changeLanguage = (newLocale: string) => {
  i18n.locale = newLocale;
};

export const translate = (key: TranslationKeys): string => i18n.t(key);
