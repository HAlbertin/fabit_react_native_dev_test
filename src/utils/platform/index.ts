import { Platform } from 'react-native';

const isAndroid = () => Platform.OS === 'android';
const isIos = () => Platform.OS === 'ios';

const IOS = '3';
const ANDROID = '2';
const getPlatform = () => (PlatformUtils.isIos ? IOS : ANDROID);

const PlatformUtils = {
  isAndroid,
  isIos,
  getPlatform,
};

export default PlatformUtils;
