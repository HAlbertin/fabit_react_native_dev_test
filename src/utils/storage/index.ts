import { MMKV } from 'react-native-mmkv';
import StorageConstants from '../../constants/storage.constants';

const storage = new MMKV();

type storageKeys = keyof typeof StorageConstants;

const setItem = (keyName: storageKeys, value: string | number | boolean) => {
  storage.set(StorageConstants[keyName], value);
};

const getItem = (keyName: storageKeys) => storage.getString(StorageConstants[keyName]);

const deleteItem = (keyName: storageKeys) => storage.delete(keyName);

const StorageUtils = {
  setItem,
  getItem,
  deleteItem,
};

export default StorageUtils;
