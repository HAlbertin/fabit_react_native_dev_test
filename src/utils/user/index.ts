import StorageUtils from '../storage';

const isUserLoggedIn = () => !!StorageUtils.getItem('USER_SECRET');
const userNeedsVerify = () => !!StorageUtils.getItem('USER_SESSION_VERIFY');

const UserUtils = {
  isUserLoggedIn,
  userNeedsVerify,
};

export default UserUtils;
