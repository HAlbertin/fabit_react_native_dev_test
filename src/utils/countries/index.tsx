import countries from 'i18n-iso-countries';
import de from 'i18n-iso-countries/langs/de.json';
import en from 'i18n-iso-countries/langs/en.json';
import es from 'i18n-iso-countries/langs/es.json';
import fr from 'i18n-iso-countries/langs/fr.json';
import it from 'i18n-iso-countries/langs/it.json';
import pt from 'i18n-iso-countries/langs/pt.json';
import Flag from '../../components/flag';
import { getCurrentLanguage } from '../language';

const __setConfig = () => {
  countries.registerLocale(en);
  countries.registerLocale(es);
  countries.registerLocale(pt);
  countries.registerLocale(it);
  countries.registerLocale(fr);
  countries.registerLocale(de);
};

const GetAllCountries = () => {
  const countriesObj = countries.getNames(getCurrentLanguage(), { select: 'official' });
  const countriesList = Object.entries(countriesObj).map(e => ({
    key: e[0],
    value: e[1],
    icon: <Flag isoCode={e[0]} />,
  }));
  return countriesList;
};

__setConfig();

const CountriesUtils = {
  GetAllCountries,
};

export default CountriesUtils;
