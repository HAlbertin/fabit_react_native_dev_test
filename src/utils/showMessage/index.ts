import { Alert } from 'react-native';
import { translate } from '../language';
import { TranslationKeys } from '../language/translations/translations.interface';

const showMessage = (title: TranslationKeys, message?: TranslationKeys) =>
  Alert.alert(translate(title), message ? translate(message) : '');

const ShowMessageUtils = {
  showMessage,
};

export default ShowMessageUtils;
