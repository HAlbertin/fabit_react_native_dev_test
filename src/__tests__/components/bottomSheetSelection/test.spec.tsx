import BottomSheetSelection from '../../../components/bottomSheetSelection';
import { renderWithProviders } from '../../../utils/test';

describe('<BottomSheetSelection />', () => {
  const itemListMock = [
    {
      key: '1',
      value: 'teste',
    },
  ];

  it('should match snapshot', () => {
    const { toJSON } = renderWithProviders(
      <BottomSheetSelection onPress={jest.fn()} show={true} itemList={itemListMock} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
