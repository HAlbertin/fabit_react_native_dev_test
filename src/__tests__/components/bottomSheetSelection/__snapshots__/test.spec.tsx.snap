// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheetSelection /> should match snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    safeArea={0}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "elevation": 2,
          "height": 1284,
          "position": "absolute",
          "shadowOpacity": 0.2,
          "width": "100%",
        },
        Object {
          "bottom": -1334,
        },
      ]
    }
    testID="bottomsheet-animated-container"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F7F8F7",
            "borderRadius": 100,
            "height": 6,
            "marginBottom": 10,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 10,
            "width": 82,
          },
        ]
      }
      testID="bottomsheet-animated-header"
    />
    <View
      style={
        Array [
          Object {
            "paddingBottom": 0,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgb(231, 234, 231)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#000",
                "borderRadius": 4,
                "borderStyle": "solid",
                "borderWidth": 1,
                "marginTop": 8,
                "width": "100%",
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "height": 2,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "transform": Array [
                  Object {
                    "scaleY": 0.5,
                  },
                ],
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "minHeight": 62,
                },
              ]
            }
          >
            <TextInput
              autoCapitalize="none"
              autoCorrect={false}
              blurOnSubmit={false}
              editable={true}
              maxFontSizeMultiplier={1.5}
              multiline={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Search"
              placeholderTextColor="#707070"
              selectionColor="#FFB81C"
              style={
                Array [
                  Object {
                    "flexGrow": 1,
                    "margin": 0,
                  },
                  Object {
                    "paddingLeft": 44,
                    "paddingRight": 12,
                  },
                  Object {
                    "height": 62,
                  },
                  Object {
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  false,
                  Object {
                    "marginLeft": 36,
                    "paddingLeft": 8,
                  },
                ]
              }
              testID="search-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 24,
                  "justifyContent": "center",
                  "position": "absolute",
                  "width": 24,
                },
                Object {
                  "left": 12,
                  "top": 19,
                },
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text />
              </View>
            </View>
          </View>
        </View>
        <Text
          color="#E02020"
          style={
            Array [
              Object {
                "color": "#E02020",
                "fontSize": 16,
              },
            ]
          }
          testID="search-input-error"
        >
          
        </Text>
      </View>
      <RCTScrollView
        data={
          Array [
            Object {
              "key": "1",
              "value": "teste",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        testID="list-select-flatlist"
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View>
              <View
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    false,
                    Array [
                      Object {
                        "padding": 8,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="list-item-1"
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "marginVertical": 6,
                          "paddingLeft": 8,
                        },
                        Object {
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "fontSize": 16,
                            },
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      teste
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.12)",
                      "height": 0.5,
                    },
                    undefined,
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
